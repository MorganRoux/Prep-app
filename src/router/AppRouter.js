
import React from 'react'
import DashBoardPage from '../components/DashboardPage';
import LoginPage from '../components/LoginPage';
import {Router, Switch} from 'react-router-dom';
import PublicRoute from '../router/PublicRoute';
import PrivateRoute from '../router/PrivateRoute';
import createHistory from 'history/createBrowserHistory';

export const history = createHistory();

const AppRouter = () => (
    <Router history={history}>
    <div>
        <Switch>
            <PublicRoute path="/" component={LoginPage} exact={true} />
            <PrivateRoute path="/dashboard" component={DashBoardPage}/>
        </Switch>
        </div>
    </Router>
);

//export const AppRouter = () => (routes);

export default AppRouter;