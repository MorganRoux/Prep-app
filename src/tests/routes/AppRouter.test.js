import AppRouter from '../../router/AppRouter';
import { shallow } from 'enzyme';
import React from 'react'

test('should render AppRouter' , () => {
    const wrapper = shallow(<AppRouter />);
    expect(wrapper).toMatchSnapshot();
})