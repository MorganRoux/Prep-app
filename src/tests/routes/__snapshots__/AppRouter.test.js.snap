// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AppRouter 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div>
    <Switch>
      <Connect(PublicRoute)
        component={[Function]}
        exact={true}
        path="/"
      />
      <Connect(PrivateRoute)
        component={[Function]}
        path="/dashboard"
      />
    </Switch>
  </div>
</Router>
`;
